### PR TITLE
postmark 1.4 with most if not all the interfaces

### PR DESCRIPTION
**Pull request by Dirrk**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://postmarkapp.com/developer/api/overview
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by benbayard — CHANGES_REQUESTED** [review]
This appears to be a fundamental re-write, almost if it was started from scratch.

Did version 1.4 really have breaking changes like these? 

It appears the major change is the inclusion of promises. 

I'd personally like to see this in two commits: 

1. Wrapping things in the namespace
2. Adding in promises. 

The naming convention in this file is directly taken from postmark's naming.

**Comment by RyanCavanaugh** [comment]
@benbayard do you feel this is OK to merge now?

**Comment by benbayard** [comment]
I'm still waiting on answers to the questions. This PR is hard to follow, so without replies from OP it's hard to allow this in with breaking changes.

**Comment by RyanCavanaugh** [comment]
OK, if you could file a "fresh" code review with Requested Changes status, that'd help the bot. Thanks!

**Review by weswigham — CHANGES_REQUESTED** [review]
Waiting on responses to @benbayard 's comments.

**Comment by RyanCavanaugh** [comment]
@Dirrk Please address comments from the code reviewers.

**Comment by RyanCavanaugh** [comment]
@benbayard - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Comment by minestarks** [comment]
I'm closing this PR as stale since requested revisions have never been addressed. If you would still like to get these changes in, please open a new PR. Thank you!
